### PR TITLE
Show possible Metal NER

### PR DESCRIPTION
**Pull request by kwalcock**

### CODE REVIEW

**Comment by kwalcock** [comment]
This will not be ready the merging for quite some time.  I just find it very convenient to discuss proposed changes in a PR.

@vikas95 , given that I'm not barely aware of what you're up to, here are changes that might make its integration possible.  See eidos.conf for where the model files might be placed.  Then see EidosProcessor.newRecognizeNamedEntities.  This may or may not be the right place to put it.  I was hoping that whatever goes here would be compatible with what happens in GeoNormFinder.  If not, one or both should be changed.  In other words, they probably need to be coordinated.  HTH.  You might update this branch with your progress.

**Comment by MihaiSurdeanu** [comment]
LGTM Keith. Thanks a lot!
I don't see the resource file name for the CompactWordEmbeddings. I am assuming you use the one with the frequency threshold > 10?

**Comment by kwalcock** [comment]
I only made the minimal changes in class names to get it to compile, so it is still using the full glove embeddings file.  I'd make the file name change in a separate PR so that it's more obvious what the cause is for changes in output.  Perhaps there's also a grounding test that I can rerun.

File: src/main/scala/org/clulab/wm/eidos/EidosProcessor.scala [hunk 1]
```diff
@@ -60,6 +63,24 @@ class EidosEnglishProcessor(val language: String, cutoff: Int) extends FastNLPPr
...
+  // TODO Vikas
+  def newRecognizeNamedEntities(doc: Document): Unit = {
+    doc.sentences.foreach { sentence =>
+      val words = sentence.words
+      val posTags = Some(sentence.tags.get.toIndexedSeq) // these are probably wrong
+      val neTags = Some(sentence.norms.get.toIndexedSeq) // these are probably wrong
+      val annotatedSentence = new AnnotatedSentence(words, posTags, neTags)
+      val predictions = metal.predict(0, annotatedSentence)
+
+      sentence.entities = Some(predictions.toArray) // this is probably wrong
```
Comment (MihaiSurdeanu) [inline src/main/scala/org/clulab/wm/eidos/EidosProcessor.scala:81]: @vikas95: this should work.

**Comment by MihaiSurdeanu** [comment]
Also, @vikas95: please add unit test(s) to make sure that the LOC NER produces the expected LOC entities on several sentences.

**Comment by kwalcock** [comment]
Just a heads up...  It was my impression that the test was failing (prematurely) because the model wasn't available.  The PR was holding, waiting for it to be added.  There may be other problems, but I ran the code up to that point and then couldn't continue, at least not yet.

**Comment by kwalcock** [comment]
@vikas95, that ${EidosSystem.path} is defined earlier in the file, so "/org/clulab/wm/eidos/english" and the files start in src/main/resources.  In total, it describes files in "src/main/resources/org/clulab/wm/eidos/english".

```
EidosSystem {
  language = english
  domain = wm
  path = /org/clulab/${EidosSystem.domain}/eidos/${EidosSystem.language}
```

**Comment by kwalcock** [comment]
@vikas95, it is loading the model, so that is integrated.  The code in newRecognizeNamedEntities seems at first glance to be wrong.  This is where I was just guessing.  You can probably fix it.  Run any of these tests to see what's going wrong:
```
[error] Error during tests:
[error] 	org.clulab.wm.eidos.text.english.raps.TestRaps
[error] 	org.clulab.wm.eidos.serialization.obj.TestSerialization
[error] 	org.clulab.wm.eidos.serialization.jsonld.TestJLDFormat
[error] 	org.clulab.wm.eidos.system.TestEntityFinder
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc5
[error] 	org.clulab.wm.eidos.serialization.jsonld.TestJLDSerializer
[error] 	org.clulab.wm.eidos.text.english.raps.TestRaps1
[error] 	org.clulab.wm.eidos.text.english.cag.TestCags
[error] 	org.clulab.wm.eidos.system.TestGlobalAction
[error] 	org.clulab.wm.eidos.system.TestParallel
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc8
[error] 	org.clulab.wm.eidos.text.english.cag.TestCagP1
[error] 	org.clulab.wm.eidos.text.english.cag.TestExtraText
[error] 	org.clulab.wm.eidos.serialization.TestDocSerialization
[error] 	org.clulab.wm.eidos.text.TestImplicitText
[error] 	org.clulab.wm.eidos.text.english.cag.TestCagP0
[error] 	org.clulab.wm.eidos.serialization.txt.TestTxtSerialization
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc2
[error] 	org.clulab.wm.eidos.text.english.cag.TestCagP4
[error] 	org.clulab.wm.eidos.system.TestHedging
[error] 	org.clulab.wm.eidos.text.TestDocumentFilter
[error] 	org.clulab.wm.eidos.system.TestEidosTokenizer
[error] 	org.clulab.wm.eidos.serialization.jsonld.TestJLDDeserializer
[error] 	org.clulab.wm.eidos.text.TestUnicode
[error] 	org.clulab.wm.eidos.text.english.cag.TestEidosMention
[error] 	org.clulab.wm.eidos.serialization.json.TestJSONSerializer
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc3
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc6
[error] 	org.clulab.wm.eidos.text.english.cag.TestCagP3
[error] 	org.clulab.wm.eidos.system.TestSerial
[error] 	org.clulab.wm.eidos.system.TestNegation
[error] 	org.clulab.wm.eidos.rule.TestJointAdjectives
[error] 	org.clulab.wm.eidos.system.TestCorefHandler
[error] 	org.clulab.wm.eidos.system.TestDomainOntology
[error] 	org.clulab.wm.eidos.system.TestCachedDomainOntology
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc1
[error] 	org.clulab.wm.eidos.system.TestGeonorm
[error] 	org.clulab.wm.eidos.system.TestAttachmentHandler
[error] 	org.clulab.wm.eidos.system.TestSaxon
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc4
[error] 	org.clulab.wm.eidos.text.english.grounding.TestGrounding
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc0
[error] 	org.clulab.wm.eidos.system.TestEidosProcessor
[error] 	org.clulab.wm.eidos.test.TestTest
[error] 	org.clulab.wm.eidos.text.english.cag.TestCagP6
[error] 	org.clulab.wm.eidos.groundings.TestOntologyGrounder
[error] 	org.clulab.wm.eidos.text.english.cag.TestCagP2
[error] 	org.clulab.wm.eidos.utils.TestMentionUtils
[error] 	org.clulab.wm.eidos.system.TestEidosActions
[error] 	org.clulab.wm.eidos.text.english.cag.TestCagP5
[error] 	org.clulab.wm.eidos.system.TestFiltering
[error] 	org.clulab.wm.eidos.text.english.eval6.TestDoc7
[error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
```

**Comment by MihaiSurdeanu** [comment]
@vikas95: please make sure that unit tests pass. Thanks!

**Comment by kwalcock** [comment]
Vikas and I met and In the meantime I've been tentatively convinced that my code is in the wrong place.  It depends on a lot on what the model has been trained to do, but it sounds like it better matches things happening in GeoNormFinder.find().  He's checking out some things and if they work, this PR will be abandoned.  If someone knows better, don't hesitate to set me straight.

**Comment by MihaiSurdeanu** [comment]
What do you have in mind?

**Comment by vikas95** [comment]
I have replaced the old GeoLocationExtractor (https://github.com/clulab/geonorm/blob/08bcaf05f55ff2806cde40d5df569be872f50459/src/main/scala/org/clulab/geonorm/GeoNorm.scala#L37) with the MetalNER ensuring that it generates the same output format (entity_word_start_indexes and entity_word_end_indexes).

I will commit the local changes in a new branch and start a pull request from there (suggested by @kwalcock).

**Comment by kwalcock** [comment]
Perhaps I had better ask.  If this new metal version is meant to deal with all sorts of named entities and replace the NER from FastNLPProcessor, then EidosEnglishProcessor is a good place to put it.  If it only deals with locations and replaces or patches up the results from FastNLPProcessor, then it might better be in GeoNormFinder or GeoNorm.  I was thinking it did the former and that performing NER twice would not be good.

**Comment by MihaiSurdeanu** [comment]
at least short term, it's the latter.

**Comment by kwalcock** [comment]
@vikas95 or @vikasy stand by for a PR in which some things are moved around.  Thanks.

**Pull request closed without merging**